Packages/com.unity.inputsystem/InputSystem/Plugins/XR/Haptics/GetHapticCapabilitiesCommand.cs

```diff
@@ -1,31 +1,104 @@
 // ENABLE_VR is not defined on Game Core but the assembly is available with limited features when the XR module is enabled.
 #if UNITY_INPUT_SYSTEM_ENABLE_XR && (ENABLE_VR || UNITY_GAMECORE) || PACKAGE_DOCS_GENERATION
+using System;
 using System.Runtime.InteropServices;
 using UnityEngine.InputSystem.LowLevel;
 using UnityEngine.InputSystem.Utilities;
 
 namespace UnityEngine.InputSystem.XR.Haptics
 {
-    public struct HapticCapabilities
+    /// <summary>
+    /// Describes the haptic capabilities of a specific device.
+    /// </summary>
+    public readonly struct HapticCapabilities
     {
-        public HapticCapabilities(uint numChannels, uint frequencyHz, uint maxBufferSize)
```

Comment:
Thanks for this PR, Chris!

This seems to be a breaking change for the package and we would need to maintain this method. A bit annoying to have to leave it as we won't do a 2.0 soon. But maybe there's a way to fix the issue you found without removing this constructor?

Reply:
The constructor isn't removed, I just moved it under the new one. It has the same signature and parameter names, so it should be backwards compatible.

Reply:
Ah, my bad. Should have looked better in the diff. Thanks and great to know 👍